lib/provider/downloader_provider.dart

```diff
@@ -383,6 +435,26 @@ class DownloadProvider extends ChangeNotifier {
 
     _downloads.add(downloadItem);
     notifyListeners();
+
+    final downloadDir = Directory(savePath);
+    if (!await downloadDir.exists()) {
+      try {
+        await downloadDir.create(recursive: true);
+      } catch (e) {
+        log('Error creating download directory: $e', name: 'DownloadProvider');
+        return null;
+      }
+    }
+
```

Comment:
This should not be necessary, as the downloader should handle it. At this point, the download is already queued, so an error will occur as soon as it attempts to write the metadata. Which ist after this step